src/frontend/src/lib/api/signer.api.ts

```diff
@@ -86,6 +87,17 @@ export const sendBtc = async ({
 	return sendBtc(params);
 };
 
+export const getSchnorrPublicKey = async ({
+	identity,
+	...rest
+}: CanisterApiFunctionParams<{ derivationPath: string[]; keyId: SchnorrKeyId }>): Promise<
+	Uint8Array | number[]
+> => {
+	const { getSchnorrPublicKey } = await signerCanister({ identity });
+
+	return await getSchnorrPublicKey(rest);
+};
+
```

Comment:
It would be nice to be able to test that this gives the same result as a command line call such as:
```
# Authorize payment
dfx canister call cycles_ledger icrc2_approve "
  record {
    amount = 10000000000000;
    spender = record {
      owner = principal \"${SIGNER_CANISTER_ID}\";
    };
  }"

# Get the public key of the web user
dfx canister call signer schnorr_public_key --with-cycles 1T --wallet $WALLET '(
  record {
    key_id = record { algorithm = variant { ed25519 }; name = "dfx_test_key" };
    canister_id = opt principal "the-web-principal";
    derivation_path = vec {
      blob "SOL";
      blob "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00";
    };
  },
  opt variant { CallerPaysIcrc2Cycles },
)'
```

Reply:
I'll deploy this branch locally and check.

Reply:
Once this is available in the UI I can check.  The derivation path should be at least ["SOL" ] and probably something like ["SOL", "testnet/mainnet", subaccount_number], with the subaccount number set to `[0u8;32]`.